Common/include/toolboxes/CQuasiNewtonDriver.hpp

```diff
@@ -0,0 +1,253 @@
+/*!
+ * \file CQuasiNewtonDriver.hpp
+ * \brief Implements a method to accelerate and stabilize the convergence
+ * of fixed point iterations, the history of past iterates is used to compute
+ * a least squares approximation of the inverse of the Jacobian, which is then
+ * used to correct the natural solution of the fixed point iteration.
+ * \note Based on the IQN-ILS method, see DOI 10.1007/s11831-013-9085-5 and
+ * references therein.
+ * \author P. Gomes
+ * \version 7.0.5 "Blackbird"
+ *
+ * SU2 Project Website: https://su2code.github.io
+ *
+ * The SU2 Project is maintained by the SU2 Foundation
+ * (http://su2foundation.org)
+ *
+ * Copyright 2012-2020, SU2 Contributors (cf. AUTHORS.md)
+ *
+ * SU2 is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * SU2 is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with SU2. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include "../omp_structure.hpp"
+#include "../mpi_structure.hpp"
+#include "CSymmetricMatrix.hpp"
+
+/*!
+ * \class A quasi-Newton fixed-point (FP) accelerator based on IQN-ILS.
+ * \note The implementation prioritizes storage, the LS problem is solved
+ * via the normal equations as that is easy to make parallel over MPI, it
+ * may however be unstable with large sample sizes (compared to QR decomp).
+ * Usage: Allocate, store the initial solution (operator (i,j), default is 0),
+ * run the FP, store its result ("FPresult"), compute new solution, use it
+ * as the new input of the FP, run the FP, etc.
+ */
+template<class Scalar_t>
+class CQuasiNewtonDriver {
+public:
+  using Scalar = Scalar_t;
+  using Index = typename su2matrix<Scalar>::Index;
+  static_assert(std::is_floating_point<Scalar>::value,"");
+
+private:
+  using MPI_Wrapper = typename SelectMPIWrapper<Scalar>::W;
+
+  enum: size_t {BLOCK_SIZE = 1024};     /*!< \brief Loop tiling parameter. */
+  std::vector<su2matrix<Scalar> > X, R; /*!< \brief Input and residual history of the FP. */
+  su2matrix<Scalar> corr;               /*!< \brief Correction to the natural FP result. */
```

Comment:
Just at one intermediate point it's the correction to the natural FP result, right? It first holds the current solution, then the delta of corrected solutions, then the corrected solution. That should be noted somewhere (so that it also goes well together with the `FPresult()` code comment) or reflected in the name.

Reply:
Called it work now and updated the comment.